extractors/bookseries/karelians/extractors/forest_extractor.py

```diff
@@ -0,0 +1,26 @@
+import re
+
+import core.utils.regex_utils as regexUtils
+from core.pipeline_construction.base_extractor import BaseExtractor
+
+
+class ForestExtractor(BaseExtractor):
+    extraction_key = 'metsää'
+
+    def __init__(self, cursor_location_depends_on=None, options=None):
+        super(ForestExtractor, self).__init__(cursor_location_depends_on, options)
+        self.FOREST_PATTERN = r'(?:([0-9][0-9]) ha metsää){s<=1}|(?<!H)(a metsää){s<=1}|(metsää [0-9][0-9] ha){s<=1}'

```

Comment:
`\d` is the same as writing `[0-9]`, FYI

Reply:
```suggestion
        self.FOREST_PATTERN = r'(?:([0-9][0-9])\sha\smetsää){s<=1}|(?<!H)(a\smetsää){s<=1}|(metsää\s[0-9][0-9]\sha){s<=1}'

```
The OCR software that has gone through the scanned books sometimes produces typos, as we discussed. Those typos can even concern `' '` characters (spaces). It's safer to try to match those with `\s`

Reply:
What's with this negative lookahead? `(?<!H)(a\smetsää){s<=1}` It's trying to capture "a metsää", allowing for one typo, and not allowed to be preceded by the character H? Interesting case. Is it the same as "ha metsää" but using a different unit?

Reply:
Generally speaking, these patterns could use some comments that explain what they are, or more importantly, why they are written the way they are. You can look at the comments for some of the other extractors.

Reply:
In addition of comment about some why some "special patterns" are used (for example the one about 'H' character in this case) it might be a good idea to add unit test which tests the particular case and in the process gives an example when the said pattern is useful/required.